apps/daimo-mobile/src/logic/notify.ts

```diff
@@ -125,3 +117,60 @@ class PushNotificationManager {
     this.accountManager.setCurrentAccount({ ...acc, pushToken: token.data });
   }
 }
+
+interface NotificationsAccess {
+  permission: Notifications.PermissionResponse | undefined;
+  ask: () => Promise<void>;
+}
+
+export function useNotificationsAccess(): NotificationsAccess {
+  const [permission, setPermission] = useState<
+    Notifications.PermissionResponse | undefined
+  >();
+
+  const fetchPermissions = async () => {
+    await Notifications.getPermissionsAsync().then(setPermission);
+  };
+
+  useEffect(() => {
+    fetchPermissions();
+  }, []);
+
+  const ask = async () => {
+    // Refresh permission state before prompting.
+    const latestPermission = await Notifications.getPermissionsAsync();
+
+    if (!latestPermission.granted) {
+      await requestNotificationsAccess(latestPermission.canAskAgain);
+    }
+
+    fetchPermissions();
+    getPushNotificationManager().maybeSavePushTokenForAccount();
+  };
+
+  return { permission, ask };
+}
+
+// Ask for push notifications access, either via a direct prompt or by asking
+// to open system settings.
+async function requestNotificationsAccess(canAskAgain: boolean) {
+  if (!Device.isDevice) {
+    console.log(`[NOTIFY] not on a real device, skipping notification request`);
+    Alert.alert(
+      "Push notifications not supported in simulator",
+      undefined,
+      [
+        {
+          text: `Continue`,
+        },
+      ],
+      { cancelable: false }
+    );
+    return;
+  }
```

Comment:
this section might not be necessary

Reply:
good call yeah looks like its not -- I was just keeping it from old code, dropped.